### PR TITLE
API code submission for FLIP fest 'Playground Feature: Markdown READMEs 18' 

### PR DESCRIPTION
**Pull request by hichana**
Closes: N/A

## Description
This PR comprises the work by @j00lz and @hichana on the FLIP fest  '[Playground Feature: Markdown READMEs 18](https://github.com/onflow/flip-fest/issues/18)'. It adds a 'PROJECT' section that allows the user to create a title, description and readme for the project, with functionality integrated into the export feature (ex. the readme.md file exports with the rest of the code) and various redirects schemes to enhance the user experience. 

The API supports the new functionality, with included tests to make sure the new data members are persisted to the API.

We made a comprehensive guide with screenshots to illustrate the features: https://github.com/hichana/flip-fest/blob/submissions/issue-%2318/milestone-%232/TeamExponential/submission/issue-18/milestone-2/TeamExponential/README.md

______

For contributor use:

- [x ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md))
- [x ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- x[ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md).
- [n/a ] Updated relevant documentation 
- [x ] Re-reviewed `Files changed` in the Github PR explorer
- [n/a ] Added appropriate labels

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -12,6 +12,8 @@ type Project {
   publicId: UUID!
   parentId: UUID
   title: String!
+  description: String!
```
Comment (MaxStalker) [inline schema.graphql:15]: Will this solution work for existing projects, where this field is empty? :thinking:

Comment (justjoolz) [reply schema.graphql:15]: I don't believe backwards compatibility or migrating old data was mentioned as a requirement for this flip.

Comment (MaxStalker) [reply schema.graphql:15]: Correct me if I am wrong, but you are saying that if we merge this PR and then deploy it all projects created prior this moment will stop working? 🤔

Comment (hichana) [reply schema.graphql:15]: @MaxStalker sorry if there is any misunderstanding here. I believe simply making the title, description and readme as not required fields, so we remove the exclamation point, then old projects will work. Should we check with Pete about this?

Comment (MaxStalker) [reply schema.graphql:15]: @justjoolz is right - FLIP description doesn't have the requirement for migration/compatibility.
We can handle this before merging this PR into master.

Personally, I was just wondering if you considered how your code would affect old deployed projects.

Comment (hichana) [reply schema.graphql]: Marking this complete @MaxStalker -- 'title', 'description' and 'readme' are now optional data members in the schema with my last commit and based on our chat about this. Marking this as resolved, tho happy to circle back after Pete provides guidance.

**Review by MaxStalker — APPROVED** [review]
Thank you for participating in discussion and amendments to the code!

**Review by sideninja — APPROVED** [review]

File: playground_test.go [hunk 2]
```diff
@@ -106,18 +110,24 @@ type GetProjectResponse struct {
 }
 
 const MutationUpdateProjectPersist = `
-mutation($projectId: UUID!, $persist: Boolean!) {
-  updateProject(input: { id: $projectId, persist: $persist }) {
+mutation($projectId: UUID!, $title: String!, $description: String!, $readme: String!, $persist: Boolean!) {
+  updateProject(input: { id: $projectId, title: $title, description: $description, readme: $readme, persist: $persist }) {
     id
+		title
```
Comment (janezpodhostnik) [inline playground_test.go:116]: There is a tabs vs. spaces issue here. I suggest converting to spaces, to match the lines above and below.

File: model/project.go [hunk 3]
```diff
@@ -170,6 +183,17 @@ func (p *InternalProject) Save() ([]datastore.Property, error) {
...
 	}
 
+	// blueMonday policy building: https://github.com/microcosm-cc/bluemonday#policy-building
+	bmUSC := bluemonday.UGCPolicy()
+	bmUSC.AllowImages()
+	bmUSC.AllowAttrs("src").OnElements("img")
+
+	bmStrict := bluemonday.StrictPolicy()
+
+	sanitizedTitle := bmStrict.Sanitize(p.Title)
```
Comment (janezpodhostnik) [inline model/project.go:193]: If it is not too much trouble I would really like to see a test where the description/title/readme is sanitized.

**Review by janezpodhostnik — APPROVED** [review]
Looks good, and a really nice addition. I left two minor comments, if you want to address them.

**Pull request merged**